src/components/ItemList.svelte

```diff
@@ -9,31 +9,45 @@
   import FilterInput from "./FilterInput.svelte";
   import Markdown from "./Markdown.svelte";
 
+  import { isExpired } from "../state/metrics";
+
   let DEFAULT_ITEMS_PER_PAGE = 20;
 
   export let appName;
   export let items;
   export let itemType;
-  export let filteredItems;
-  export let totalItems;
-  export let paginatedItems;
+
+  export let filterText = "";
+
+  let showExpired = false;
+  let filteredItems = items.filter((item) => !isExpired(item.expires));
+  let pagedItems;
 
   let currentPage = writable(1);
   setContext("currentPage", currentPage);
 
-  totalItems = items.length;
-  paginatedItems = chunk([...items], DEFAULT_ITEMS_PER_PAGE);
+  const goToPage = (page) => {
```

Comment:
I took out the pagination logic from the old filter function and put in a separate function to make the code a bit easier to read, hopefully…